src/pages/home/settings/DeleteAccountDialog.tsx

```diff
@@ -0,0 +1,126 @@
+  useGetProfileQuery,
+} from "modules/session";
+
+const DeleteAccountDialog: FunctionComponent = () => {
+  const deleteAccountPhrase = "YES";
+  const regExpPhrase = new RegExp(`^${deleteAccountPhrase}$`);
+
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const { data: { id } = emptyProfile } = useGetProfileQuery();
+
+  const [deleteAccount, isLoading] = useDeleteAccountThunk();
+
+  const initialValues = {
+    confirmationStatement: "",
+  };
+
+  const validationSchema = Yup.object().shape({
+    confirmationStatement: Yup.string()
+      .required(`Must type the word "${deleteAccountPhrase}"`)
+      .matches(regExpPhrase, `Must type the word "${deleteAccountPhrase}"`),
+  });
+
+  const handleSubmit = ({ confirmationStatement }: FormikValues) => {
+    if (confirmationStatement === deleteAccountPhrase) {
+      deleteAccount({ id });
+      setIsModalOpen(false);
+    }
+  };
+
+  const handleCloseDialog = () => {
+    setIsModalOpen(false);
+  };
+
+  const handleOpenDialog = () => {
+    setIsModalOpen(true);
+  };
+
+  return (
+    <>
+      <Button onClick={ handleOpenDialog } variant="secondary" color="magazine">
+        Delete account
+      </Button>
+      <Dialog open={ isModalOpen }>
+        <Formik
+          initialValues={ initialValues }
+          onSubmit={ handleSubmit }
+          validationSchema={ validationSchema }
+        >
+          { () => (
+            <Form>
+              <Box
+                sx={ {
+                  display: "flex",
+                  justifyContent: "center",
+                  flexDirection: "column",
+                  p: 3,
+                  rowGap: 2,
+                } }
+              >
+                <Stack sx={ { rowGap: 1 } }>
+                  <Typography variant="body2">
+                    You can still turn back...
+                  </Typography>
+                  <Typography variant="subtitle1">
+                    Once you delete your profile it&apos;s gone for good. If
+                    you&apos;re sure and would like to proceed, please confirm
+                    by typing &ldquo;YES&rdquo; below.
+                  </Typography>
+                </Stack>
+                <Stack
+                  sx={ {
+                    display: "flex",
+                    justifyContent: "start",
+                  } }
+                >
+                  <TextInputField name="confirmationStatement"></TextInputField>
+                </Stack>
+
+                <Stack
+                  sx={ {
+                    display: "flex",
+                    flexDirection: "row",
+                    justifyContent: "end",
+                    alignItems: "end",
+                    columnGap: 1.5,
+                    pt: 1.5,
+                  } }
+                >
+                  <Button
+                    variant="secondary"
+                    color="music"
+                    width="compact"
+                    onClick={ handleCloseDialog }
```

Comment:
In a state where `isLoading` is true, would you want to disable the cancel button so they dont exit early, making them think they canceled the process? just some thoughts 🤔

Reply:
I think the thought flew through my head, but didn't add it on first pass. Will add it before I merge.